integrations/terraform/hashicorp-vault/README.md

```diff
@@ -0,0 +1,219 @@
+# Materialize Terraform Provider + HashiCorp Vault
+
+HashiCorp Vault is a powerful tool for securely accessing secrets. This demo will guide you through the process of setting up and using [Vault](https://www.vaultproject.io/), as well as demonstrating how to interact with the Materialize Terraform provider using the `materialize_secret` resource with secrets stored in Vault.
+
+## Overview
+
+![](https://github.com/MaterializeInc/demos/assets/21223421/edc48e99-77b3-4c47-8e86-472c51d45f70)
+
+## Prerequisites
+
+- Install [Vault](https://developer.hashicorp.com/vault/downloads)
+- Install [Terraform](https://developer.hashicorp.com/terraform/downloads) 1.0.3 or later
+- [Materialize Cloud](https://cloud.materialize.com/) account.
+- Follow the steps in the [README.md](../README.md) file to set up your AWS configuration and Materialize details.
+
+## Setting up Vault
+
+Before we can use Vault, we need to start a Vault server. For testing purposes, we'll run Vault in dev mode:
+
+```bash
+vault server -dev
+```
+
+This will start Vault in dev mode and output an Unseal Key and a Root Token.
+
+Make sure to note the Root Token as we'll use it to authenticate with Vault.
+
+## Interacting with Vault
+
+Open a new terminal and export the `VAULT_ADDR`:
+
+```bash
+export VAULT_ADDR='http://127.0.0.1:8200'
+```
+
+You'll need to authenticate yourself with the root token:
+
+```bash
+vault login <Root-Token>
+```
+
+Once authenticated, you can begin to interact with Vault.
+
+## Adding and Retrieving a Secret in Vault
+
+To store a secret in Vault, we can use the `vault kv put` command.
+
+```bash
+vault kv put secret/materialize pgpass=some-secret-value
+```
+
+The vault `kv put` command creates a key-value secret at the path specified. In this case, the secret is stored at `secret/materialize`.
+
+To retrieve a secret, you use the `vault kv get` command:
+
+```bash
+vault kv get secret/materialize
+```
+
+## Integrating Vault with Terraform:
+
+Terraform has a Vault provider which you can use to retrieve secrets from Vault.
+
+First, let's start by creating a file called `main.tf` and then in that file you will need to add the Vault provider and the Materialize provider:
+
+```hcl
+terraform {
+  required_providers {
+    vault = {
+      source = "hashicorp/vault"
+      version = "~> 3.15"
+    }
+    materialize = {
+      source = "MaterializeInc/materialize"
+      version = "0.0.5"
+    }
+  }
+}
+```
+
+You'll then have to initialize the Vault provider:
+
+```hcl
+provider "vault" {
+  address = "http://localhost:8200"
+  token = "<Root-Token>"
+}
+```
+
+Replace `<Root-Token>` with the actual Root Token of your Vault server.
+
+After that initialize the Materialize provider:
+
+```hcl
+provider "materialize" {
+  host     = local.materialize_host
```

Comment:
For this and the other demo in this repo should we use `var` instead of `local` to be consistent with the provider and the modules?